ui/src/authors/containers/AssignOneDifferentProfileContainer.jsx

```diff
@@ -0,0 +1,47 @@
+import { connect } from 'react-redux';
+
+import {
+  setPublicationSelection,
+  setPublicationsClaimedSelection,
+  setPublicationsUnclaimedSelection,
+  clearPublicationSelection,
+  clearPublicationsClaimedSelection,
+  assignDifferentProfileClaimedPapers,
+  assignDifferentProfileUnclaimedPapers,
+  clearPublicationsUnclaimedSelection,
+  clearPublicationsCanNotClaimSelection,
+  setPublicationsCanNotClaimSelection,
+} from '../../actions/authors';
+import AssignDiffetentProfileAction from '../components/AssignDifferentProfileAction';
+
+export const stateToProps = (state) => ({
+  currentUserId: state.user.getIn(['data', 'recid']),
+});
+
+export const dispatchToProps = (
+  dispatch,
+  { recordId, disabledAssignAction, canClaimDifferentProfile }
+) => ({
+  onAssign({ from, to }) {
+    dispatch(clearPublicationSelection());
+    dispatch(clearPublicationsClaimedSelection());
+    dispatch(clearPublicationsUnclaimedSelection());
+    dispatch(clearPublicationsCanNotClaimSelection());
+    dispatch(setPublicationSelection([recordId], true));
+    if (!canClaimDifferentProfile) {
+      dispatch(setPublicationsCanNotClaimSelection([recordId], true));
+    }
+    if (!disabledAssignAction && canClaimDifferentProfile) {
+      dispatch(setPublicationsUnclaimedSelection([recordId], true));
+      dispatch(assignDifferentProfileUnclaimedPapers({ from, to }));
+    } else {
+      dispatch(setPublicationsClaimedSelection([recordId], true));
+      dispatch(assignDifferentProfileClaimedPapers({ from, to }));
```

Comment:
If this https://github.com/inspirehep/inspirehep/pull/2429/files#diff-d05b30a07ecd69ef8dd8c77c9f48af79e1759380f12ea2f4e7aec7218e5c2dd1R31-R33 happens it will also run this else, is it what we want?

Reply:
yes cause paper can be both claimed and have incompatible author name